source/maze/MouseController.cpp

```diff
@@ -187,6 +187,50 @@ void MouseController::moveNextCell()
 	}
 }
 
+void MouseController::setUnsearchDes(int n)
+{
+	srand(101);
+	int count = 0;
+	//count non searched cells
+	for (int i = 0; i < CONFIG_MAX_ROW_SIZE; i++) {
+		for (int j = 0; j < CONFIG_MAX_COL_SIZE; j++) {
+			if (getCell(Position{i,j}).status == unsearched) {
+				count++;
+			}
+		}
+	}
+	//find destinations
+	while (destinations.size() < n && count > 0) {
+		//choose random row
+		int temp = (int) (rand()%16);
+	    int i = temp;
+	    //look for non searched cells in row
+		for (int j = 0; j < CONFIG_MAX_COL_SIZE; j++) {
+			if (getCell(Position{i, j}).status == unsearched) {
+				if (positionIsDestination(Position{i,j})) {
+					continue;
+				}
+				destinations.push_back(Position{i,j});
+				count--;
+				break;
+			}
+	    }
```

Comment:
This is good for now...But we may have to find more cleaver way. Anyway it is not the highest priority.